src/sentences/utils.rs

```diff
@@ -316,4 +316,70 @@ mod tests {
         let result = parse_magnetic_variation("12,Q");
         assert!(result.is_err());
     }
+
+    #[test]
+    fn test_parse_array_string() {
+        let result = array_string::<5>("12345");
+        assert!(result.is_ok());
+        let result = array_string::<5>("123456");
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_parse_number_in_range() {
+        let result = parse_number_in_range::<u8>("12", 10, 20);
+        assert!(result.is_ok());
+        let result = parse_number_in_range::<u8>("9", 10, 20);
+        assert!(result.is_err());
+        let result = parse_number_in_range::<u8>("21", 10, 20);
+        assert!(result.is_err());
```

Comment:
It would be best to validate the returned error.
Right now the function might be returning a different that the `out of range` error.